### PR TITLE
fix arm compile error

### PR DESCRIPTION
**Pull request by LHT129**
fixed: #238 
- add arm test on ci

### CODE REVIEW

File: src/simd/avx512.cpp [hunk 1]
```diff
@@ -12,8 +12,9 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-
+#if defined(ENABLE_AVX512)
 #include <immintrin.h>
```
Comment (jiaweizone) [inline src/simd/avx512.cpp:16]: Ok, fix ARM compile issue

File: .circleci/config.yml [hunk 2]
```diff
@@ -8,13 +8,23 @@ workflows:
             branches:
               ignore:
                 - main
+      - pull-request-check-aarch64:
+          filters:
+            branches:
+              ignore:
+                - main
```
Comment (jiaweizone) [inline .circleci/config.yml:15]: Why main branch need skip ARM platform check ?

File: src/simd/avx512.cpp (hunk 1 above)
Comment (LHT129) [reply src/simd/avx512.cpp:16]: yeah

File: .circleci/config.yml (hunk 2 above)
Comment (LHT129) [reply .circleci/config.yml:15]: pull request ignore main, if merged another workflow run

File: .circleci/config.yml [hunk 3]
```diff
@@ -33,12 +43,46 @@ jobs:
             - ./build
       - run: make test_parallel
 
+  pull-request-check-aarch64:
+    docker:
+      - image: ubuntu:22.04
+    resource_class: arm.medium
+    steps:
+      - checkout
+      - run: apt update && apt install -y gfortran python3-dev libomp-15-dev gcc make cmake g++
```
Comment (wxyucs) [inline .circleci/config.yml]: replace with the dependencies install script in scripts/deps/

Comment (LHT129) [reply .circleci/config.yml]: done

**Review by wxyucs — APPROVED** [review]
lgtm

**Review by jiaweizone — APPROVED** [review]
LGTM

**Review by ShawnShawnYou — APPROVED** [review]

**Pull request merged**